docs/usage-notes.md

```diff
@@ -0,0 +1,52 @@
+This document shows several usage scenarios of SMACK that require special flags.
+
+## Bitwise Operations and Integer Casts
+If the program to verify contains bitwise operations or integer casts, then the flag `--bit-precise` may be required. The reason is that SMACK uses the SMT theory of integers to encode machine integers by default where bitwise operations are encoded using uninterpreted functions returning arbitrary values. Furthermore, special encoding is required to handle integer signedness casts, which is not also enabled automatically.
+
+The following program demonstrate the problems in the presence of bitwise operations.
+
+### Example
+```C
+#include "smack.h"
+
+int main () {
+  unsigned y = __VERIFIER_nondet_unsigned_int();
+  assume (y < 4U);
+  y >>= 2U;
+  assert (!y);
+}
+```
+This program should verify. However, if the we enable SMACK in its default mode, an error would be reported.
```

Comment:
This program should verify. However, if we run SMACK in its default mode, it reports an assertion violation.